README.md

```diff
@@ -77,6 +77,36 @@ pip install tensorflow
 
 In use cases where you require a small footprint, such as on a raspberry pi or similar embedded devices, you will want to install the TFLite Interpreter. You can install it for your platform by following [the instructions here](https://www.tensorflow.org/lite/guide/python#install_just_the_tensorflow_lite_interpreter).
 
+### Speech Pipeline
+
+The Speech Pipeline is the component that ties together VAD, Wakeword, and ASR. Essentially, this component listens to a frame of audio to determine if speech is present. If speech is detected, the Wakeword model processes the subsequent frames of audio looking for the specific keyword. If the keyword is found, the pipeline is activated and converts the following audio into a transcript. The Speech Pipeline is established like this from the previous initialized components:
+
+```python
+from spokestack.io.pyaudio import PyAudioInput
+from spokestack.pipeline import SpeechPipeline
+from spokestack.vad.webrtc import VoiceActivityTrigger
+from spokestack.asr.spokestack.speech_recognizer import SpeechRecognizer
+
+mic = PyAudioInput()
+vad = VoiceActivityTrigger()
+asr = SpeechRecognizer("spokestack_id", "spokestack_secret")
+
+
+pipeline = SpeechPipeline(mic, [vad, wake, asr])
+pipeline.start()
+pipeline.run()
+```
+
+#### Pipeline Callbacks
+
+Pipeline callbacks allow additional code to be executed when a speech event is detected. For example, we can print when the pipeline is activated by registering a function with the `pipeline.event` decorator.
+
+```python
+@pipeline.event
+def on_activate(context):
+    print(context.is_active)
+```
+
 ### Audio Input
```

Comment:
Now that the structure has been flipped, there's actually little need to go over initialization of the individual components since it's all in the first section. You could have a "Pipeline components" section where you give some more details on each component, or you could get rid of the individual component explanations altogether in favor of some more detail on pipeline callbacks, which are what people need to know about in order to get the ASR transcripts. Right now, there are instructions on how to set up the pipeline, but not too much on how to make use of it.

If you do keep the sections on individual components, they should at least be demoted a header level or two so they're not as prominent as the pipeline, NLU, and TTS.